### PR TITLE
feat: check if ens was already associated to snapshot space

### PR DESCRIPTION
**Pull request by deep-quality-dev**

### CODE REVIEW

**Review by Whytecrowe — APPROVED** [review]

File: src/snapshot-io/index.ts [hunk 1]
```diff
@@ -147,23 +147,22 @@ class SnapshotClient {
...
-      name: item.name,
-      avatar: item.avatarUri,
-      network: item.network,
-      admins: item.admins,
-      period: item.voting.period ? item.voting.period : undefined,
-      strategies: item.strategies,
-      followers: item.followers,
-    }));
+    this.spaces = list.map(
+      (item: any): SnapshotSpace => ({
```
Comment (RyanCross) [inline src/snapshot-io/index.ts:157]: Nit: Try and avoid any in production code if you can.

**Review by RyanCross — APPROVED** [review]

File: src/snapshot-io/index.ts (hunk 1 above)
Comment (deep-quality-dev) [reply src/snapshot-io/index.ts:157]: I want too, but snapshot sdk is javascript project, defined as any

Comment (JamesEarle) [reply src/snapshot-io/index.ts:157]: @deep-quality-dev add a comment above to mark this as intentional to the linter `eslint-disable-next-line @typescript-eslint/no-explicit-any`

**Comment by JamesEarle** [comment]
@deep-quality-dev Please resolve conflicts

File: src/snapshot-io/index.ts [hunk 2]
```diff
@@ -185,15 +184,16 @@ class SnapshotClient {
...
+      proposals: item.proposalsCount ? Number(item.proposalsCount) : 0,
       followers: item.followersCount,
       admins: item.admins,
       strategies: item.strategies,
       threshold: item.filters.minScore,
+      duration: item.voting.period,
       delay: item.voting.delay,
       quorum: item.voting.quorum,
-    };
+    } as SnapshotSpaceDetails;
```
[the remark this answers is not in the document]
Comment (deep-quality-dev) [reply src/snapshot-io/index.ts:196]: Yes, with `as T`, I want to make sure I defined all the fields of `SnapshotSpaceDetails` in returning value.

**Pull request merged**